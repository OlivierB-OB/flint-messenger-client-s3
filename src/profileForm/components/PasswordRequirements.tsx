import Grid from '@material-ui/core/Grid';
import React from 'react';
import { PasswordRequirement } from './PasswordRequirement';
import { IPasswordField } from '../types';

export interface IPasswordRequirementsProps {
  password: IPasswordField;
}

export function PasswordRequirements({ password }: IPasswordRequirementsProps) {
  const { hasLower, hasUpper, hasNumber, hasSymbol, hasValidLength } = password;
  return (
    <Grid container direction="column" alignContent="flex-start" style={{ margin: '1rem 0' }}>
      <PasswordRequirement check={hasLower} message="password contains some lowercase characters..." />
      <PasswordRequirement check={hasUpper} message="password contains some uppercase characters..." />
      <PasswordRequirement check={hasNumber} message="password contains some numbers..." />
      <PasswordRequirement check={hasSymbol} message="password contains some symbols..." />
      <PasswordRequirement check={hasValidLength} message="password contains 8..30 characters..." />
    </Grid>
  );
}
