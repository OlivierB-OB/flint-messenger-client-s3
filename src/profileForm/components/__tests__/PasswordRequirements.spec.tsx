import React from 'react';
import { shallow } from 'enzyme';
import { PasswordRequirements } from '../PasswordRequirements';

describe('PasswordRequirement', () => {
  it('should display all password requirements with their states', async () => {
    const password: any = {
      hasLower: 'hasLower value',
      hasUpper: 'hasUpper value',
      hasNumber: 'hasNumber value',
      hasSymbol: 'hasSymbol value',
      hasValidLength: 'hasValidLength value',
    };
    const component = shallow(<PasswordRequirements password={password} />);
    expect(component).toMatchSnapshot();
  });
});
