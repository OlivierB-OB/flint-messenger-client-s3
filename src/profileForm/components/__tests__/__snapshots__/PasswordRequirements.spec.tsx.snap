// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PasswordRequirement should display all password requirements with their states 1`] = `
<WithStyles(ForwardRef(Grid))
  alignContent="flex-start"
  container={true}
  direction="column"
  style={
    Object {
      "margin": "1rem 0",
    }
  }
>
  <PasswordRequirement
    check="hasLower value"
    message="password contains some lowercase characters..."
  />
  <PasswordRequirement
    check="hasUpper value"
    message="password contains some uppercase characters..."
  />
  <PasswordRequirement
    check="hasNumber value"
    message="password contains some numbers..."
  />
  <PasswordRequirement
    check="hasSymbol value"
    message="password contains some symbols..."
  />
  <PasswordRequirement
    check="hasValidLength value"
    message="password contains 8..30 characters..."
  />
</WithStyles(ForwardRef(Grid))>
`;
